dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java

```diff
@@ -482,6 +484,27 @@ private static Set<String> getVerifiedPlatforms(Set<? extends Version> workflowV
         return platforms;
     }
 
+    private static List<String> getDistinctDescriptorTypeVersions(Entry entry, Set<? extends Version> workflowVersions) {
+        String language;
+        if (entry instanceof Tool && ((Tool)entry).getDescriptorType().size() == 1) {
+            // Only set descriptor type versions if there's one descriptor type otherwise we can't tell which version belongs to which type without looking at the source files
+            language = ((Tool)entry).getDescriptorType().get(0);
+        } else if (entry instanceof BioWorkflow || entry instanceof AppTool) {
+            language = ((Workflow)entry).getDescriptorType().toString();
+        } else {
+            return List.of();
+        }
+
+        // Get a list of unique descriptor type versions with the descriptor type prepended. Ex: 'WDL 1.0'
+        List<String> descriptorTypeVersions = workflowVersions.stream()
```

Comment:
It would be good if we didn't have to load all the versions to calculate this, and calculate this with JQL instead. But we already load all the versions on lines 316 and 317, so it's only worth it if those get fixed as well.

Perhaps a followup ticket and/or todo?